src/main/java/service/OptionCli.java

```diff
@@ -0,0 +1,113 @@
+package service;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.cli.*;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Slf4j
+@Data
+public class OptionCli {
+    private String[] arguments;
+    private String dataTime;
+    private CommandLineParser parser;
+    private Options options;
+    private CommandLine cmd;
+
+    public OptionCli(String[] arguments) {
+        this.arguments = arguments;
+        this.parser = new DefaultParser();
+        this.options = optionsBuild();
+        try {
+            this.cmd = parser.parse(this.options, this.arguments);
+        } catch (ParseException e) {
+            log.info("Invalid values ​​for parse... \n\tFor help, type -h or --help");
+        }
+    }
+
+    private Options optionsBuild() {
+        this.options = new Options();
+
+        this.options.addOption(Option.builder("h")
+                .longOpt("help")
+                .hasArg(false)
+                .desc("Print this message")
+                .required(false)
+                .build());
+
+        this.options.addOption(Option.builder("C")
+                .longOpt("current")
+                .hasArg(false)
+                .desc("Set this to view the current weather")
+                .required(false)
+                .build());
+
+        this.options.addOption(Option.builder("F")
+                .longOpt("forecast")
+                .argName(" \"yyyy-MM-dd HH:mm:ss\" ")
+                .hasArg(true)
+                .optionalArg(true)
+                .desc("Set this to view the forecast *no parameter will display available dates and forecast times")
+                .required(false)
+                .build());
+
+        this.options.addOption(Option.builder("c")
+                .longOpt("city")
+                .argName(" name ")
+                .hasArg(true)
+                .desc("Enter the city name")
+                .required(false)
+                .build());
+
+        this.options.addOption(Option.builder("g")
+                .longOpt("geo")
+                .argName(" latitude longitude ")
+                .hasArg(true)
+                .numberOfArgs(2)
+                .desc("Search by geographic coordinates")
+                .required(false)
+                .build());
+
+        return this.options;
+    }
+
+    public void helpBuild() {
+        HelpFormatter formatter = new HelpFormatter();
+        formatter.printHelp("Options ", optionsBuild(), true);
+    }
+
+    public String optionBuildQuery() {
+        //Current
+        if (cmd.hasOption("C") && cmd.hasOption("c")) {
+            StringBuilder stringBuilder = new StringBuilder(cmd.getOptionValue("c"));
+            for (String arg : cmd.getArgList()) {
+                stringBuilder.append("+").append(arg);
+            }
+            return new QueryUrlFactory("C").buildQueryByNameCity(stringBuilder.toString());
+        }
```

Comment:
linijki 82-89 robią to samo co linijki 97-104 tylko dla literki "C" zamiast "F" + ustawiają date z "F"
na pewno da się to uprościć i zostawić tylko jedną z tych 2.